gsr_booking/views.py

```diff
@@ -267,27 +267,22 @@ def invites(self, request, pk):
             ).data
         )
 
-    @action(detail=True, methods=["post"], url_path="book-room")
-    def book_room(self, request, pk):
-        request.data.update({"group_id": pk})
+    @action(detail=True, methods=["post"], url_path="book-rooms")
+    def book_rooms(self, request, pk):
         booking_serialized = GroupBookingRequestSerializer(data=request.data)
         if not booking_serialized.is_valid():
             return Response(status=400)
 
         booking_data = booking_serialized.data
 
-        group = get_object_or_404(Group, pk=booking_data["group_id"])
+        group = get_object_or_404(Group, pk=pk)
+
+        # must be admin of the group to book
         if not group.has_member(request.user) or not group.has_admin(request.user):
```

Comment:
A way to do this in one database call instead of two would be to query for the GroupMembership for this group and user, and then check that that membership `is_admin`.